### ISSUE TITLE
Reduce duplicate posts - track posted titles

### ISSUE THREAD

**Issue by catleeball** [catleeball/tmnt_wikipedia_bot#9]
Fill a sqlite database with which titles have been posted already to avoid duplication.

  1. Backfill database with existing posts
  3. When candidate title is found, check db if it has already been posted
  2. On successful new post, add post to db

Proposed table columns:
- title (str) (indexed)
- twitter post id (int)
- mastodon post id
- stresses (str)
- timestamp

**Comment by Samwalton9** [comment]
Is this necessary? It doesn't seem to me that it would be a problem to re-post a title give the seemingly large pool; it's probably unlikely that a title be posted more than once, right? And if it is likely that we re-post, then this would run into a problem where we've posted most titles and start to not be able to find a new one in a reasonable timeframe.

**Comment by catleeball** [comment]
I was afraid of exhausting the candidate pool too, but added this issue since anecdotally a lot of users have complained about repeats. I'll check the log file of its stdout and see what the frequency of repeats looks like, but honestly might close this bug soon for the reasons you mentioned. 🙂

**Comment by Samwalton9** [comment]
Data would be helpful, for sure!

**Comment by catleeball** [comment]
Here's the data for title frequencies for all the bot's posts up to ~Jun 5: https://docs.google.com/spreadsheets/d/1QZW5fIF-SoKrLKogI2MoVZ-yJFP1bwkGtv778btLVxQ/edit

There were a fair number of repeats! When I kicked the bot into gear again, I removed filtering titles with parenthesis to add more variety.

I'm planning to slowly rewrite the bot in rust for funzies and learning, and when I do I may change the title picking logic. Instead, I plan to use a wikipedia data dump and iterate all candidate titles, maybe in lexical order. We'll see if I get around to it eventually. 🙂 

Anyways, I'll close this bug out! Thanks for the feedback!
